source/plugin/wgen/customwgen.rst

```diff
@@ -0,0 +1,251 @@
+
+
+Creating Custom Base Terrain
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Changing the base ``GenerationPopulator`` of a world generator allows you to change the base terrain shape
+generation of the world. A generator populator will roughly follow the procedure of using the seed and biome
+information to seed and modify a series of noise maps, from which the terrain is formed. The terrain created
+in a modified base generator populator should only consist of stone blocks, to allow the biomes to properly replace
+blocks for biome-specific ground cover.
+
+.. code-block:: java
+
+	public class SinusoidalGenerator implements GenerationPopulator {
+		@Override
+		public void populate(World world, MutableBlockVolume buffer, ImmutableBiomeArea biomes) {
+			for(int x = buffer.getBlockMin().getX(), x < buffer.getBlockMax().getX(); x++) {
+				for(int z = buffer.getBlockMin().getZ(), z < buffer.getBlockMax().getZ(); z++) {
+					BiomeType biome = biomes.get(x, z);
+					int height = getHeight(x, z, biome);
+					for(int y = 0; y < height || y < 64; y++) {
+						if(y < height) {
+							buffer.set(x, y, z, BlockTypes.STONE);
+						} else {
+							buffer.set(x, y, z, BlockTypes.WATER);
+						}
+					}
+				}
+			}
+		}
+
+		private int getHeight(int x, int z, BiomeType biome) {
+			double sx = Math.sin(x/64d)+1;
+			double sz = Math.sin(z/64d)+1;
+			double value = (sx + sz) / 4d;
+			BiomeGenerationSettings settings = biome.getGenerationSettings();
+			double heightRange = settings.getMaxHeight() - settings.getMinHeight();
+			double height = heightRange * value | settings.getMinHeight;
+			return GenericMath.floor(height * 256);
+		}
+	}
+
+This is a fairly simple example of a base terrain generation populator (at least, if you look past the math to
+calculate the height). For each column in the buffered area we want to calculate a height value, and then fill
+in everything below that with stone and leave everything above it as air (or water if we're still below sea-level).
+
+[You can of course ... TBA]
+
+[TBA Add some simple example of creating noise with flow-noise and creating terrain from it.]
+
+
+Creating Custom GenerationPopulators
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[TBA]
+
+
+Creating Custom Populators
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Custom populators can be used to add a great variety of custom features. To create a custom populator you need
+only create a class implementing the Populator interface and add it to the list of populators attached to a
+biome, or to a world generator if you want it applied globally.
+
+The key thing to remember when creating a populator is that the area affected by the populator is a 16x16 area
+offset by 8 in both the x and z axes
+
+[see image]
+
+[Insert an example or two.]
+
+Creating Custom Biomes
+~~~~~~~~~~~~~~~~~~~~~~
+
+*At present it is not possible to create entirely new biomes from purely within the SpongeAPI.*
+
+
+[TODO]
```

Comment:
Oh heh this was my todo for the world gen branch that somehow made it into that gist.